src/diet.ts

```diff
@@ -392,16 +315,164 @@ export function runDiet(): void {
-      eat(1, $item`fudge spork`);
+  if (have($item`astral six-pack`)) {
+    use($item`astral six-pack`);
+  }
+
+  fillShotglass();
+
+  if (
+    get("barrelShrineUnlocked") &&
+    !get("_barrelPrayer") &&
+    $classes`Turtle Tamer, Accordion Thief`.includes(myClass())
+  ) {
+    cliExecute("barrelprayer buff");
+  }
+
+  const { bestSpleenItem } = getBestSpleenItems();
+  const embezzlers = embezzlerCount();
+  if (embezzlers) {
+    if (mySpleenUse() < spleenLimit()) {
+      if (!have($effect`Eau d' Clochard`)) {
+        if (!have($item`beggin' cologne`)) {
+          const cologne = new Potion($item`beggin' cologne`);
+          const equilibriumPrice = cologne.gross(embezzlers) - valuePerSpleen(bestSpleenItem);
+          if (equilibriumPrice > 0) buy(1, $item`beggin' cologne`, equilibriumPrice);
+        }
+        if (have($item`beggin' cologne`)) {
+          chew(1, $item`beggin' cologne`);
+        }
+      }
     }
-    MayoClinic.setMayoMinder(MayoClinic.Mayo.zapine, 1);
-    eatSafe(1, $item`jumping horseradish`);
+    if (have($skill`Sweet Synthesis`) && mySpleenUse() < spleenLimit()) {
+      ensureEffect($effect`Synthesis: Collection`);
+    }
+    if (have($item`body spradium`) && mySpleenUse() < spleenLimit()) {
+      ensureEffect($effect`Boxing Day Glow`);
+    }
+  }
+
+  const liverStomachCleaners = $items`spice melange, distention pill, synthetic dog hair pill, cuppa Voraci tea, cuppa Sobrie tea`;
+  const helpers = [Mayo.flex, $item`Special Seasoning`, saladFork, frostyMug];
+  const haveMayo = getWorkshed() === $item`portable Mayo Clinic`;
+  const horseradishLength = haveMayo ? 100 : 50;
+  // Adjust value of horseradish if we have mayo, since we're using Mayozapine instead of Mayoflex.
+  const horseradishValue =
+    horseradishLength * baseMeat +
+    Math.min(horseradishLength, embezzlers) * 750 -
+    (haveMayo ? MPA : 0);
+  const menu = [
+    // BOOZE
+    new MenuItem($item`dirt julep`, {
+      maximum: 1,
+      additionalValue: 40 * baseMeat + Math.min(embezzlers, 40) * 750,
+    }),
+    new MenuItem($item`Ambitious Turkey`, {
+      maximum: 1,
+      additionalValue: 0.25 * (50 * baseMeat + Math.min(embezzlers, 50) + 750),
+    }),
+    new MenuItem($item`Ambitious Turkey`, { additionalValue: 0.25 * 50 * baseMeat }),
+    new MenuItem($item`jar of fermented pickle juice`),
+    new MenuItem($item`Sacramento wine`),
+    new MenuItem($item`splendid martini`),
+    new MenuItem($item`punch-drunk punch`),
+    new MenuItem($item`blood-red mushroom wine`),
+    new MenuItem($item`buzzing mushroom wine`),
+    new MenuItem($item`complex mushroom wine`),
+    new MenuItem($item`overpowering mushroom wine`),
+    new MenuItem($item`smooth mushroom wine`),
+    new MenuItem($item`swirling mushroom wine`),
+    new MenuItem($item`Hodgman's blanket`),
+
+    // FOOD
+    new MenuItem($item`jumping horseradish`, {
+      maximum: 1,
```

Comment:
You said that it filled you up on horseradish? That is probably a bad sign given this

Reply:
Reworked this code